pennylane/qnode_new/decorator.py

```diff
@@ -47,30 +57,41 @@ def QNode(
               TensorFlow ``tfe.Variable`` objects.
 
         mutable (bool): whether the QNode circuit is mutable
-        differentiable (bool): whether the QNode is differentiable
+        diff (str, None): the method of differentiation to use in the created QNode.
+
+            * ``"best"``: Best available method. Uses the device directly to compute

```

Comment:
Maybe an Enum class could be created for these values?

Reply:
For user-facing features, it is generally more user friendly to use strings (as they do not need to be imported, and they add additional complexity to the UI). However, for developer-facing signatures, enums are usually better.

Reply:
Coming from strongly typed languages like C#, this is also something I struggled with for a long time. But the main benefit of using enums in strongly typed languages is that you get a compile-time check that the value you input makes sense. As you can't have that in python there is actually no big advantage in using enums in python. But I still like it for developer-facing code because it is cleaner!

Reply:
Yep, although they provide some minor advantages, they seem to have no major upside indeed. One pitfall I could see is a user typing in something else than what they intended and then experiencing unwanted behavior. But this could be amended by having the if-else statements adjusted well.

Reply:
Good point, I'll add in some input validation

Reply:
Yeah, if you use strings then the last `else` should raise a `TypeError('unknown option ...')`